javagen/src/main/java/com/azure/autorest/Javagen.java

```diff
@@ -137,121 +137,124 @@ protected NodeTuple representJavaBeanProperty(Object javaBean, Property property
     protected JavaPackage writeToTemplates(CodeModel codeModel, Client client, JavaSettings settings,
                                            boolean generateSwaggerMarkdown) {
         JavaPackage javaPackage = new JavaPackage(this);
-        // Service client
-        if (CoreUtils.isNullOrEmpty(client.getServiceClients())) {
-            javaPackage.addServiceClient(client.getServiceClient().getPackage(),
-                client.getServiceClient().getClassName(), client.getServiceClient());
-        } else {
-            // multi-client from TypeSpec
-            for (ServiceClient serviceClient : client.getServiceClients()) {
-                javaPackage.addServiceClient(serviceClient.getPackage(), serviceClient.getClassName(), serviceClient);
+        if (client.getServiceClient() != null || !CoreUtils.isNullOrEmpty(client.getServiceClients())) {
```

Comment:
Doesn't `CoreUtils.isNullOrEmpty` handle the nullness check?

Reply:
`CoreUtils.isNullOrEmpty()` is only for collections or string. Not for null checking an object.